src/seedu/addressbook/data/person/Person.java

```diff
@@ -83,5 +83,18 @@ public int hashCode() {
     public String toString() {
         return getAsTextShowAll();
     }
+    
+    public String getPrintableString() {
+    	return getPrintableString(name, phone, email, address);
+    }
+    
+    public String getPrintableString(Printable...printables){
+    	StringBuilder builder = new StringBuilder();
+    	for (Printable p : printables){
+    		builder.append(p.getPrintableString());
+    		builder.append(" ");
```

Comment:
Doesn't this result in an extra space at the end? You could have used `StringJoiner` or `String.join` instead.